### PR TITLE
fix(textfield): textarea actually grows with multiline

### PR DESCRIPTION
**Pull request by blunteshwar**
## Description

sp-textarea was not growing when typing multiple lines of text, so I updated the`height` of textarea with `scrollHeight`

## Related issue(s)
#4259 

-

## Motivation and context

It updates the height of textarea with attributes {multiline,grows} when typing multiple lines of text.

## How has this been tested?

-   [ ] _Test case 1_
    1. Go [here](https://opensource.adobe.com/spectrum-web-components/components/textarea/#grows)
    2. Remove any text from the middle of the textarea
    3. Start adding text and press Enter
    4. Check the textarea doesn't grow 

## Screenshots (if appropriate)

https://github.com/adobe/spectrum-web-components/assets/32033946/5ba144e0-c84a-4fde-951f-b29a79f1af0d

## Types of changes

-   [x] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)
-   [ ] Chore (minor updates related to the tooling or maintenance of the repository, does not impact compiled assets)

## Checklist

-   [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
-   [x] My code follows the code style of this project.
-   [x] If my change required a change to the documentation, I have updated the documentation in this pull request.
-   [x] I have read the **[CONTRIBUTING](<(https://github.com/adobe/spectrum-web-components/blob/main/CONTRIBUTING.md)>)** document.
-   [ ] I have added tests to cover my changes.
-   [x] All new and existing tests passed.
-   [x] I have reviewed at the Accessibility Practices for this feature, see: [Aria Practices](https://www.w3.org/TR/wai-aria-practices/)

## Best practices

This repository uses conventional commit syntax for each commit message; note that the GitHub UI does not use this by default so be cautious when accepting suggested changes. Avoid the "Update branch" button on the pull request and opt instead for rebasing your branch against `main`.

### CODE REVIEW

File: packages/textfield/src/Textfield.ts [hunk 1]
```diff
@@ -249,6 +249,8 @@ export class TextfieldBase extends ManageHelpText(
             }
         }
         this.value = this.inputElement.value;
+        this.inputElement.style.height = 'auto';
+        this.inputElement.style.height = `${this.inputElement.scrollHeight}px`;
```
Comment (Westbrook) [inline packages/textfield/src/Textfield.ts:253]: Please look closer at making this possible via CSS. Relying on a paint requisite measurement on every keystroke feels like overkill here.

Possibly:
```
white-space: pre-line;
    min-height: max-content;
    overflow: clip;
```

In association with better managed whitespace in the actual `render()` method for the sizing text?

File: packages/textfield/src/textfield.css [hunk 2]
```diff
@@ -134,5 +134,7 @@ textarea {
     left: 0;
     height: 100%;
     resize: none;
-    overflow: hidden;
+    white-space: pre-line;
+    min-height: max-content;
+    overflow: clip;
```
Comment (Westbrook) [inline packages/textfield/src/textfield.css:139]: Did you mean to maybe put this at line 54?

Comment (blunteshwar) [reply packages/textfield/src/textfield.css]: I tried these css changes but they didn't work, neither on line 137 nor on line 54.
In both cases the textarea was not growing at all!

Comment (Westbrook) [reply packages/textfield/src/textfield.css]: There is a CSS answer for this. Please find it.

Comment (blunteshwar) [reply packages/textfield/src/textfield.css]: Apparently `white-space: pre-line;`  is sufficient enough for this

**Comment by Westbrook** [comment]
Currently ends up with textarea always one line longer than the actual text:
image

**Comment by Rajdeepc** [comment]
@blunteshwar lets try to add a re-review request after a full VRT test pass to save time of the review process?

**Comment by blunteshwar** [comment]
> @blunteshwar lets try to add a re-review request after a full VRT test pass going forward to save time of the review process?

Sure! Will keep in mind from next time

File: packages/textfield/test/textfield.test.ts [hunk 3]
```diff
@@ -323,6 +323,34 @@ describe('Textfield', () => {
...
+            el.focus();
+            el.select();
+            for (let i = 0; i < 100; i++) {
+                await sendKeys({
+                    type: 'ab',
+                });
+                await sendKeys({ press: 'Enter' });
+            }
+            const finalHeight = textArea.offsetHeight;
+            expect(initialHeight).to.not.equal(finalHeight);
```
Comment (Westbrook) [inline packages/textfield/test/textfield.test.ts:350]:
```suggestion
            expect(initialHeight).to.be.lt(finalHeight);
```
More specific to test what we want to happen not just that _something_ happened.

File: packages/textfield/src/textfield.css [hunk 4]
```diff
@@ -52,6 +52,8 @@ textarea {
     block-size: auto;
     word-break: break-word;
     opacity: 0;
+    white-space: pre-line;
+    overflow: clip;
```
Comment (Westbrook) [inline packages/textfield/src/textfield.css:56]: Is this required with the change in the `render` method? Less is always more 😉

File: packages/textfield/src/Textfield.ts [hunk 5]
```diff
@@ -287,13 +287,12 @@ export class TextfieldBase extends ManageHelpText(
     protected get displayValue(): string {
         return this.value.toString();
     }
-
+    // prettier-ignore
```
Comment (Westbrook) [inline packages/textfield/src/Textfield.ts:290]:
```suggestion

    // prettier-ignore
```
Nit.

File: packages/textfield/src/textfield.css (hunk 4 above)
Comment (blunteshwar) [reply packages/textfield/src/textfield.css]: `white-space: pre-line;` is still requied, but yeah we can remove `overflow: clip;`

**Review by Westbrook — APPROVED** [review]
LGTM! :shipit:

**Pull request merged**